src/coreclr/minipal/Unix/doublemapping.cpp

```diff
@@ -198,8 +198,10 @@ bool VMToOSInterface::ReleaseDoubleMappedMemory(void *mapperHandle, void* pStart
 {
 #ifndef TARGET_OSX
     int fd = (int)(size_t)mapperHandle;
-    mmap(pStart, size, PROT_READ | PROT_WRITE, MAP_SHARED | MAP_FIXED, fd, offset);
-    memset(pStart, 0, size);
+    if (mmap(pStart, size, PROT_READ | PROT_WRITE, MAP_SHARED | MAP_FIXED, fd, offset) != MAP_FAILED)
```

Comment:
This would hide a fatal error. The `mmap` should never fail unless there was a memory corruption or a bug in the code using this method. If we just didn't fill the memory and unmapped it, when we would later allocate the same block of shared memory, but it would not be zeroed. The executable heaps that are using this code or code using memory allocated from those heaps would then fail in various ways.
It would be better to return `false` if it fails and also change the caller in executableallocator.cpp to check the return code and call the g_fatalErrorHandler if this failed.

Reply:
Thank you for review.
I fixed it.